src/gui/plugins/plotting/GazeboPlotting.hh

```diff
@@ -0,0 +1,81 @@
+#ifndef IGNITION_GUI_PLUGINS_PLOTTING_HH_
+#define IGNITION_GUI_PLUGINS_PLOTTING_HH_
+
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/qt.h>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include "ignition/gazebo/components/Name.hh"
+#include "ignition/gazebo/components/Pose.hh"
+#include <ignition/gazebo/gui/GuiSystem.hh>
+#include <string>
+#include <iostream>
+#include <QTimer>
+#include "Transport.hh"
+#include <QStandardItemModel>
+
+#define TIMEOUT 100
+using namespace std;
+
+/// \brief Interface to Communicate with Qml, and to be a common code between the GuiPlugin and GazeboPlugin
+class PlottingInterface : public QObject
+{
+    Q_OBJECT
+private:
+    /// \brief responsible for accessing the transport messages (publishing and subscribing)
+    Transport* transport;
```

Comment:
A couple of style details (I just realized not all of them are in our [style guide](https://ignitionrobotics.org/docs/all/contributing#style-guides) yet):

* The `*` should be to the right, i.e. `Transport *transport`
* The `private` and `public` should be in front of every function, see for example:

    https://github.com/ignitionrobotics/ign-gazebo/blob/98d41cd7bd6a0f7c9293c0e1cbca81e59c5db0dc/src/gui/plugins/scene3d/Scene3D.hh#L79-L83